### ISSUE TITLE
kEstimateTableReadersMem metric incorrect when using partitioned index or filter

### ISSUE THREAD

**Issue by igorcanadi** [facebook/rocksdb#8979]
Our application uses partitioned index and partitioned filters. The value of kEstimateTableReadersMem metric is incredibly low, a few MB for a database that is hundreds of GB in size. From code inspection, it looks like [PartitionedFilterBlockReader:: ApproximateMemoryUsage()](https://github.com/facebook/rocksdb/blob/2acffecca1f3027bfc71e38c3cbbe9ae6b17b535/table/block_based/partitioned_filter_block.cc#L417-L426) and [PartitionIndexReader::ApproximateMemoryUsage()](https://github.com/facebook/rocksdb/blob/2acffecca1f3027bfc71e38c3cbbe9ae6b17b535/table/block_based/partitioned_index_reader.h#L33-L42) only account for the root block memory usage and not for the partitioned blocks. Is this a correct understanding?

**Comment by akankshamahajan15** [comment]
Yes, partitioned index/filters are not taking second level partitions into account.
